internal/extsvc/gitlab/client.go

```
@@ -435,16 +435,23 @@ func (e ProjectNotFoundError) NotFound() bool { return true }
 
 var MockGetOAuthContext func() *oauthutil.OAuthContext
 
-func getOAuthContext() *oauthutil.OAuthContext {
+// getOAuthContext matches the corresponding auth provider using the given
+// baseURL and returns the oauthutil.OAuthContext of it.
+func getOAuthContext(baseURL *url.URL) *oauthutil.OAuthContext {
 	if MockGetOAuthContext != nil {
 		return MockGetOAuthContext()
 	}
 
+	baseURL.Path = "" // Cut out path like "/api/v4"
+	rootURL := strings.TrimSuffix(baseURL.String(), "/")
 	for _, authProvider := range conf.SiteConfig().AuthProviders {
 		if authProvider.Gitlab != nil {
 			p := authProvider.Gitlab
-
 			glURL := strings.TrimSuffix(p.Url, "/")
+			if glURL != rootURL {
+				continue
+			}
```

Is the baseURL always ensured to have a trimmed prefix as well? Slightly scared that there can be mismatches here that remain unnoticed and we just silently return `nil` 😬

---

Hmm, "a trimmed prefix"?

---

Oh, the other way to do is something like `if strings.HasPrefix(baseURL.String(), glURL)`, is that what you're thinking of?

---

maybe? Or just compare URL.Host and URL.Port? not sure 🤔

---

OK, I made another mistake which makes in-place mutation `baseURL.Path = ""` which breaks tons of things 😂 Changed to use the prefix matching.